spec/presenters/guide_presenter_spec.rb

```diff
@@ -2,25 +2,32 @@
 
 RSpec.describe GuidePresenter, type: :presenter do
 
-  let(:file_names)   { %w<my/path/some_text.md my/path/and_more_text.md> }
-  let(:renderer)     { double("a_mock_renderer", render: "some_content") }
+  let(:file_names)   { ['my/path/some_text.md', 'my/path/and_more_text.md'] }
+  let(:renderer)     { instance_double("MarkdownRenderer", render: "some_content") }
   let(:mock_content) { "just some content.." }
 
-  let(:subject)      { described_class.new(file_names, renderer) }
+  let(:guide_presenter)      { described_class.new(file_names, renderer) }
 
   describe "#each_rendered_file" do
+    it "return result based on block for params" do
+      allow(File).to receive(:read).and_return mock_content
+
+      allow(renderer).to receive(:render).with(mock_content).and_return "some_content"
+      expect { |probe| guide_presenter.each_rendered_file(&probe) }.
+        to yield_successive_args(['some_text', 'some_content'], ['and_more_text', 'some_content'])
+    end
+
     it "passes the files basename and rendered content to the block argument" do
       allow(File).to receive(:read).and_return mock_content
 
       expect(renderer).to receive(:render).with mock_content
-      expect { |probe| subject.each_rendered_file &probe }.
-        to yield_successive_args(%w<some_text some_content>, %w<and_more_text some_content>)
+      guide_presenter.each_rendered_file { |name| name }
```

Comment:
Why this line is there? Is it testing something?

Reply:
@leoapost it's triggering the test.

Reply:
@leoapost changes submitted